cwl/src/main/scala/cwl/CwlWomExpression.scala

```diff
@@ -122,16 +122,40 @@ object InitialWorkDirFileGeneratorExpression {
 
     implicit val caseExpression: Case.Aux[Expression, InitialWorkDirFileEvaluator] = {
       at { expression =>
-        (_, unmappedParameterContext, mappedInputValues, expressionLib) => {
-          // A single expression which must evaluate to an array of Files
+        (ioFunctionSet, unmappedParameterContext, _, expressionLib) => {
+          // A single expression which should evaluate to a File or an array of Files.
           val expressionEvaluation = ExpressionEvaluator.eval(expression, unmappedParameterContext, expressionLib)
 
+          def stageFile(file: WomFile): Future[WomSingleFile] = {
+            // TODO WomFile could be a WomMaybePopulatedFile with secondary files but this code only stages in
+            // the primary file.
+            // The file should be staged to the initial work dir using the base filename.
+            val baseFileName = file.value.substring(file.value.lastIndexOf('/') + 1)
+            ioFunctionSet.copyFile(file.value, baseFileName)
```

Comment:
ToL: I wonder how this relates to the code in `SharedFileSystem` that historically was doing "localization" of input files in the call directory (and still is I think).
This is for InitialWorkDir so a bit different but we have now "localization code" both deep in CWL territory and in backend territory.
Also I wonder how this plays in PAPI ? If those files are in GCS we can't really `copyFile` them to the VM.

Reply:
I agree that it is weird having localization code here but this was [already](https://github.com/broadinstitute/cromwell/blob/develop/cwl/src/main/scala/cwl/CwlWomExpression.scala#L101) [the](https://github.com/broadinstitute/cromwell/blob/develop/cwl/src/main/scala/cwl/CwlWomExpression.scala#L116) [case](https://github.com/broadinstitute/cromwell/blob/develop/cwl/src/main/scala/cwl/CwlWomExpression.scala#L92) before these changes. These changes only add coverage for one type of directory entry that wasn't doing its localizations previously.

Reply:
As for PAPI, this system does return the new paths to the files in GCS and I believe they will get localized in to PAPI as inputs. However the copy here is broken for reasons I don't currently understand.